### PR TITLE
DM update automation post patching

### PR DESCRIPTION
**Pull request by sselvara1**
This script is a utility to update DM automatically post patching. This involves:
1. pull image from configured source
2. push image to local registry
3. run anisible playbook to refresh DM

Jira: CGTS-47365

### CODE REVIEW

File: scripts/dm_update_post_patch.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# This script automates DM patching to 
+# 1. pull image from configured source
+# 2. push image to local registry 
+# 3. run anisible playbook to refresh DM 
+
+#
+# Step-1: Set variables for the image and tags, including the reference for the Wind River repo on the Wind River Registry.
+#
+echo -e "\nStep-1: Set variables for the image and tags, including the reference for the Wind River repo on the Wind River Registry."
+export MY_MIRROR=admin-2.cumulus.wrs.com:30093
```
Comment (yjian118) [inline scripts/dm_update_post_patch.sh:20]: Note: not all the users using this registry

File: scripts/dm_update_post_patch.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+export REGISTRY_LOCAL=registry.local:9001
+
+#
+# Step-2: Set tags for the image being updated, defining the new image tags.
+#
+echo -e "\nStep-2: Set tags for the image being updated, defining the new image tags."
+export MY_IMAGE=wind-river/cloud-platform-deployment-manager
+export MY_IMAGE_TAG=WRCP_22.12-wrs.5
```
Comment (yjian118) [inline scripts/dm_update_post_patch.sh:28]: It doesn’t make sense to use version tag in the master branch

File: scripts/dm_update_post_patch.sh [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+#
+# Step-4: Pull the new image to docker, tag it, and then push it to the local registry.
+#
+echo -e "\nStep-4: Pull the new image to docker, tag it, and then push it to the local registry."
+sudo docker pull ${MY_MIRROR}/${MY_IMAGE}:${MY_IMAGE_TAG}
+export MY_LOCAL_IMAGE=docker.io/wind-river/cloud-platform-deployment-manager
+sudo docker tag ${MY_MIRROR}/${MY_IMAGE}:${MY_IMAGE_TAG} ${REGISTRY_LOCAL}/${MY_LOCAL_IMAGE}:${MY_IMAGE_TAG}
+sudo docker push ${REGISTRY_LOCAL}/${MY_LOCAL_IMAGE}:${MY_IMAGE_TAG}
+sudo crictl pull --creds "admin:Li69nux*" ${REGISTRY_LOCAL}/${MY_LOCAL_IMAGE}:${MY_IMAGE_TAG}
```
Comment (yjian118) [inline scripts/dm_update_post_patch.sh:45]: Credentials are not always the same

**Review by yjian118 — CHANGES_REQUESTED** [review]
I would suggest to abandon this commit in the DM repo. 
1 The update of the DM should use the playbook
2 Didn’t we want to leverage the sysinv to pull the new image?
3 The versions, credentials should come from the system rather than hard coded

**Comment by sselvara1** [comment]
> I would suggest to abandon this commit in the DM repo. 1 The update of the DM should use the playbook 2 Didn’t we want to leverage the sysinv to pull the new image? 3 The versions, credentials should come from the system rather than hard coded

I would suggest to abandon this commit in the DM repo.
[Susendra] I will move this commit to utility repo in starlingx project.
1 The update of the DM should use the playbook
[Susendra] sure, will replace 'helm upgrade' with 'ansible playbook'
2 Didn’t we want to leverage the sysinv to pull the new image?
[Susendra] yes, workflow diagram in HLD is updated. Will update the script to leverage sysinv to pull the new image.
3 The versions, credentials should come from the system rather than hard coded
[Susendra] ok, will get back on this.

**Comment by sselvara1** [comment]
Commit is moved to utility repo in starlingx project
https://review.opendev.org/c/starlingx/utilities/+/887154

**Pull request closed without merging**